### PR TITLE
Providing additional details of the parameters of the module creation API

### PR DESCRIPTION
**Pull request by hashitop**
## Labels

- [X] inaccuracy
- [X] clarification
- [ ] new docs
- [ ] cosmetic bug - fixing broken text or markup
- [ ] enhancement - changing the website's behavior/layout
- [ ] api - requires an update to the [changelog](https://www.terraform.io/docs/cloud/api/changelog.html)

### CODE REVIEW

File: content/source/docs/cloud/api/modules.html.md [hunk 1]
```diff
@@ -359,9 +359,9 @@ Properties without a default value are required.
 Key path                        | Type   | Default | Description
 --------------------------------|--------|---------|------------
 `data.type`                     | string |         | Must be `"registry-modules"`.
-`data.attributes.name`          | string |         | The name of this module. May contain alphanumeric characters and dashes.
-`data.attributes.provider`      | string |         | Specifies the Terraform provider that this module is used for.
-`data.attributes.namespace`     | string |         | The namespace of this module. The organization name for private modules. The namespace for public modules
+`data.attributes.name`          | string |         | The name of this module. May contain alphanumeric with dashes and underscores allowed in non-leading or trailing positions and the maximum length of 64 characters.
+`data.attributes.provider`      | string |         | Specifies the Terraform provider that this module is used for. May contain alphanumeric with the maximum length of 64 characters.
```
Comment (laurapacilio) [inline content/source/docs/cloud/api/modules.html.md:363]:
```suggestion
`data.attributes.provider`      | string |         | Specifies the Terraform provider that this module is used for. May contain alphanumeric characters. Maximum length is 64 characters.
```

File: content/source/docs/cloud/api/modules.html.md [hunk 2]
```diff
@@ -359,9 +359,9 @@ Properties without a default value are required.
...
+`data.attributes.namespace`     | string |         | The namespace of this module. The organization name for private modules. The namespace for public modules. May contain alphanumeric with dashes and underscores allowed in non-leading or trailing positions and the maximum length of 64 characters.
```
Comment (laurapacilio) [inline content/source/docs/cloud/api/modules.html.md:364]:
```suggestion
`data.attributes.namespace`     | string |         | The namespace of this module. The organization name for private modules. The namespace for public modules. May contain alphanumeric characters, with dashes and underscores allowed in non-leading or trailing positions. Maximum length is 64 characters.
```

**Review by laurapacilio — APPROVED** [review]
Reviewed for grammar, flow, etc. A couple suggestions for your review!

**Comment by hashitop** [comment]
> Reviewed for grammar, flow, etc. A couple suggestions for your review!

Thank you! took all the suggestions

**Review by joekarl — APPROVED** [review]
Thank you @hashitop!

**Pull request merged**